### ISSUE TITLE
Retrieving the Risk Metrics 

### ISSUE THREAD

**Issue by vertangelx** [quantopian/zipline#608]
Hi all, I've just started using zipline and cannot seem to figure out how to access the risk metrics (sharpe, sortino, jensen's alpha, volatility, drawdown) after running the algorithm.

Are there built in functions to get these ratios and metrics? Basically to get the same information as seen here

![image](https://cloud.githubusercontent.com/assets/8826684/8185629/781ee084-1413-11e5-8c43-f8663528655d.png)

**Comment by ssanderson** [comment]
@vertangelx how are you invoking `TradingAlgorithm`?  If you're using the iterator-style interface, then the last emitted packet will contain the "risk report" which includes most of this data.

If you're using `TradingAlgorithm.run` (which is probably the way to go), then you can add an "analyze" function that will receive a DataFrame with most of this data.  The format of that DataFrame isn't especially pleasant though: it has nested dictionaries in some of its cells.  I've written a BacktestResult object for the Quantopian research platform that better encapsulates this data, and it's on my TODO list to port it to zipline so that `TradingAlgorithm.run()` can return a nice object.  It just hasn't made it far enough up the priority queue yet.

**Comment by vertangelx** [comment]
@ssanderson Yes I'm using `TradingAlgorithm.run`, ie `perf = algo.run()`. Which part of the `perf` DataFrame can some of the ratios like sharpe ratio be found? Was digging through the `perf` DataFrame but didnt see anything related to sharpe or risk.

Btw, what is the iterator-style interface that you've mentioned? I'll probably try that too.

**Comment by ssanderson** [comment]
If you pass `SimulationParameters` to `TradingAlgorithm.__init__`, then you can do

```
for perf_packet in algorithm.get_generator():
    # Do stuff
```
This is more or less what `TradingAlgorithm.run()` does under the hood (see https://github.com/quantopian/zipline/blob/master/zipline/algorithm.py#L502), though there's some hackery to allow passing a new `SimulationParameters` when you call run().

Given that it seems reasonable to create a single `TradingAlgorithm` and then run it with different parameters, I think the run-based interface is going to become the only way to run a `TradingAlgorithm`, hopefully sooner rather than later.  I'd stick with that unless you have a reason to do otherwise.

The particular info you're looking for should be in a dictionary stored per-day in the `'cumulative_risk_metrics'` column of the `DataFrame` passed to `analyze`.  I will again make no claims that this is a pleasant or sane way of structuring the data...

**Comment by FreddieV4** [comment]
Just cross-referencing some duplicate issues to clean some stuff up.

#71, #391

**Comment by ssanderson** [comment]
@freddiev4 I think the Metrics API actually makes this problem worse, not better. The Metrics API makes it easier to customize what metrics get computed in any particular algorithm, but we still don't have a sane or documented API for getting results back out of an algorithm.

**Comment by freddiev4** [comment]
@ssanderson ah that's fair. I believe I confused this with what `AlgorithmResult` is supposed to be doing.
